app/controllers/users_controller.rb

```diff
@@ -7,9 +7,12 @@
 #= Controller for user functionality
 class UsersController < ApplicationController
   respond_to :html, :json
-  before_action :require_participating_user, only: [:enroll]
 
+  before_action :require_participating_user, only: [:enroll]
   before_action :require_signed_in, only: [:update_locale]
+  before_action :check_user_auth, only: [:index]
```

Comment:
You can use `require_admin_permissions`, defined in ApplicationController, instead.